gdal/frmts/jpegxr/jpegxrdataset.cpp

```diff
@@ -0,0 +1,1202 @@
+            "JPEGXR - unknown band %i color interpretation",
+            nBand);
+        return GCI_Undefined;
+    }
+    catch (JPEGXRException const& e)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+            "JPEGXR - Color interpretation access failed : %s", e.what());
+    }
+    return GCI_Undefined;
+}
+
+/************************************************************************/
+/*                        JPEGXRDataset()                               */
+/************************************************************************/
+
+JPEGXRDataset::JPEGXRDataset()
+    : pabyUncompressedData(nullptr)
+    , bHasUncompressed(false)
+{
+}
+
+/************************************************************************/
+/*                       ~JPEGXRDataset()                               */
+/************************************************************************/
+
+JPEGXRDataset::~JPEGXRDataset()
+{
+}
+
+/************************************************************************/
+/*                             CreateCopy()                             */
+/************************************************************************/
+
+GDALDataset* JPEGXRDataset::CreateCopy(char const* pszFilename,
+                                       GDALDataset* poSrcDS,
+                                       int bStrict,
+                                       char** papszOptions,
+                                       GDALProgressFunc /*pfnProgress*/,
+                                       void* /*pProgressData*/)
+{
+    int const nBands = poSrcDS->GetRasterCount();
+    int const nXSize = poSrcDS->GetRasterXSize();
+    int const nYSize = poSrcDS->GetRasterYSize();
+
+    /* -------------------------------------------------------------------- */
+    /*      Some some rudimentary checks                                    */
+    /* -------------------------------------------------------------------- */
+    if (nBands != 1 && nBands != 3 && nBands != 4)
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+            "JPEGXR driver doesn't support %d bands.  Must be 1 (grey), "
+            "3 (RGB) or 4 bands.\n", nBands);
+        return nullptr;
+    }
+
+    GDALDataType const eDT = poSrcDS->GetRasterBand(1)->GetRasterDataType();
+    if (eDT != GDT_Byte) // TODO
+    {
+        CPLError((bStrict) ? CE_Failure : CE_Warning, CPLE_NotSupported,
+            "JPEGCR driver doesn't support data type %s",
+            GDALGetDataTypeName(
+                poSrcDS->GetRasterBand(1)->GetRasterDataType()));
+
+        if (bStrict)
+            return nullptr;
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Collect configuration options                                   */
+    /* -------------------------------------------------------------------- */
+    JPEGXREncoderConfig config; // sets sensible defaults
+
+    if (CSLFetchNameValue(papszOptions, "QUALITY") != NULL)
```

Comment:
store the result of CSLFetchNameValue() in variable and reuse it in the next 2 occurences. Same for other options